File: 프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.py
def solution(sequence, k):
    answer = []
    left, right = 0, 0
    tmp = sequence[0]
    min_len = 1000001
    while left <= right < len(sequence):
        if tmp == k:
            if right - left + 1 < min_len:
                min_len = right - left + 1
                answer = [left, right]
            tmp -= sequence[left]
            left += 1
        elif tmp < k:
            right += 1
            if right < len(sequence):
                tmp += sequence[right]
        elif tmp > k:
            tmp -= sequence[left]
            left += 1
            
    return answer

File: 프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.js
function solution(sequence, k) {
    var answer = [];
    let left = 0, right = 0;
    let bestStart = Infinity, bestLength = Infinity;
    let sum = 0;
    
    while (right < sequence.length) {
        sum += sequence[right++];
        
        while (sum > k) {
            sum -= sequence[left++];
        }
        
        if (sum === k) {
            if (right - left < bestLength) {
                bestLength = right - left;
                bestStart = left;
                answer = [left, right - 1];
            } else if (right - left === bestLength && left < bestStart) {
                bestStart = left;
                answer = [left, right - 1];
            }
        }
    }
    return answer;
}